File: 0941-sort-array-by-parity/0941-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int l = 0;

        for(int r = 0; r < nums.length; ++r) {
            if(nums[r] % 2 == 0) {
                swap(nums, l , r);
                l++;
            }
        }

        return nums;
    }

    private void swap(int[] nums, int start, int end) {
        int temp = nums[start];
        nums[start] = nums[end];
        nums[end] = temp;
    }
}

File: 0941-sort-array-by-parity/0941-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l = -1

        for r in range(n):
            if  nums[r] % 2 == 0:
                l += 1
                nums[l], nums[r] = nums[r], nums[l]
        return nums